### PR TITLE
Add 'Move a feature along a line' example

### PR DESCRIPTION
**Pull request by jonataswalker**
This PR adds an example of how to use an encoded polyline and move a feature along it.

### CODE REVIEW

File: examples/feature-move-animation.js [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+      stopAnimation(true);
+      return;
+    }
+
+    var currentPoint = new ol.geom.Point(routeCoords[index]);
+    var feature = new ol.Feature(currentPoint);
+    vectorContext.drawFeature(feature, styles.geoMarker);
+  }
+  // tell OL3 to continue the postcompose animation
+  frameState.animate = true;
```
Comment (tschaub) [inline examples/feature-move-animation.js]: I think we should encourage folks to use [`map.render()`](http://openlayers.org/en/master/apidoc/ol.Map.html#render) here instead.

**Comment by tschaub** [comment]
Thanks for the nice example @jonataswalker.  See my one [comment above](https://github.com/openlayers/ol3/pull/4291#discussion_r43546878).  Since `frameState.animate` is not documented as part of the API, I think calling `map.render()` makes more sense here.

**Comment by jonataswalker** [comment]
@tschaub For sure, changed.

**Comment by tschaub** [comment]
This looks great @jonataswalker.  We typically ask that people squash redundant commits (force pushing to your branch to clean up history), but we can save that for next time.

Thanks for the contribution!

**Comment by jonataswalker** [comment]
My pleasure to contribute! I'll remember that (force pushing) tip.

**Comment by tschaub** [comment]
Published: http://openlayers.org/en/master/examples/feature-move-animation.html

**Pull request merged**